java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java

```diff
@@ -329,6 +329,36 @@ protected ArrowBuf read(BufferAllocator allocator, int count) throws IOException
       }
     };
 
+    BufferReader FIXEDSIZEBINARY = new BufferReader() {
+      @Override
+      protected ArrowBuf read(BufferAllocator allocator, int count) throws IOException {
+        ArrayList<byte[]> values = Lists.newArrayList();
+        int byteWidth = 0;
+        for (int i = 0; i < count; i++) {
+          parser.nextToken();
+          final byte[] value = decodeHexSafe(parser.readValueAs(String.class));
+          values.add(value);
+          if (value.length > 0) {
+            if (byteWidth == 0) {
+              byteWidth = value.length;
```

Comment:
Can you read the byteWidth from the schema rather than interpret it here?

Reply:
byteWidth info is now collected from the vector and passed in

Reply:
Sorry I am not sure I understand.. Why don't we use the byteWidth in schema to inform the expected length here?